hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoUpdateHandler.java

```diff
@@ -63,8 +106,167 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         }
     }
 
+    /**
+     * This method is called during the handle workflow. It executes the actual transaction.
+     * @param context the {@link HandleContext} which collects all information
+     * @throws HandleException if any of the checks fail
+     * @throws NullPointerException if any of the arguments are null
+     */
     @Override
-    public void handle(@NonNull final HandleContext context) throws HandleException {
-        throw new UnsupportedOperationException("Not implemented");
+    public void handle(@NonNull final HandleContext context) {
+        final var txn = requireNonNull(context).body();
+        final var op = txn.cryptoUpdateAccountOrThrow();
+        final var target = op.accountIDToUpdateOrThrow();
+
+        // validate update account exists
+        final var accountStore = context.writableStore(WritableAccountStore.class);
+        final var targetAccount = accountStore.get(target);
+        validateTrue(targetAccount != null, INVALID_ACCOUNT_ID);
+
+        // Customize the account based on fields set in transaction body
+        final var builder = updateBuilder(op, targetAccount);
+
+        // validate all checks that involve config and state
+        validateSemantics(context, targetAccount, op, builder, accountStore);
+
+        // If an account is detached and expired, validateSemantics would have thrown an exception
+        // To reach here, this update transaction has extended the expiration of the account.
+        // So, we need to set the expiredAndPendingRemoval flag to false
+        if (context.expiryValidator()
```

Comment:
I think here we just want,
```
if (targetAccount.expiredAndPendingRemoval()) {
    builder.expiredAndPendingRemoval(false);
}
```
since a valid update to an expired account must extend its expiry past current consensus time.

Reply:
Yes, that makes sense. Thanks